### PR TITLE
Add new projects: grpc-httpjson-transcoding, http_pattern_matcher

### PR DESCRIPTION
**Pull request by nareddyt**
https://github.com/envoyproxy/envoy depends on https://github.com/grpc-ecosystem/grpc-httpjson-transcoding and will soon depend on https://github.com/google/http_pattern_matcher. We want to setup OSS fuzz with these projects to adhere to Envoy extensions' [external dependency policy](https://github.com/envoyproxy/envoy/blob/main/EXTENSION_POLICY.md#extension-stability-and-security-posture). 

Both libraries have parsers that are good candidates for fuzzing. They build with bazel's rules_fuzzing and have fuzz tests already. I have https://github.com/grpc-ecosystem/grpc-httpjson-transcoding/pull/57 and https://github.com/google/http_pattern_matcher/pull/10 open to add the `.bazelrc` configurations.

Signed-off-by: Teju Nareddy <nareddyt@google.com>

### CODE REVIEW

**Comment by nareddyt** [comment]
cc @asraa from envoy's side

File: projects/grpc-httpjson-transcoding/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+declare -r QUERY='
+    let all_fuzz_tests = attr(tags, "fuzz-test", "//...") in
+    $all_fuzz_tests - attr(tags, "no-oss-fuzz", $all_fuzz_tests)
+'
+
+declare -r PACKAGE_SUFFIX="_oss_fuzz"
+declare -r OSS_FUZZ_TESTS="$(bazel query "${QUERY}" | sed "s/$/${PACKAGE_SUFFIX}/")"
+
+bazel build -c opt --config=oss-fuzz --linkopt=-lc++ \
```
Comment (asraa) [inline projects/grpc-httpjson-transcoding/build.sh:30]: i might be missing this but is `--config=oss-fuzz` meaningful?

(Envoy's are defined here https://github.com/envoyproxy/envoy/blob/ca1455ccf81ca4cd742903361d7c2a94f37f5be0/.bazelrc#L301-L314)

Comment (asraa) [reply projects/grpc-httpjson-transcoding/build.sh:30]: AH! just saw the PR description. Resolved :)

Comment (nareddyt) [reply projects/grpc-httpjson-transcoding/build.sh:30]: Yup! We are following the docs linked here: https://github.com/bazelbuild/rules_fuzzing#oss-fuzz-integration

**Comment by nareddyt** [comment]
Hi @inferno-chromium, do you have any ideas on the bazel rules_fuzzing integration failure in CI?

```
	File "/root/.cache/bazel/_bazel_root/1f2d02b1c30820fa03d68ff064b1166c/external/rules_fuzzing/fuzzing/private/oss_fuzz/repository.bzl", line 41, column 17, in _extract_build_params
		fail("Unsupported $LIB_FUZZING_ENGINE value '%s'" % fuzzing_engine_library)
Error in fail: Unsupported $LIB_FUZZING_ENGINE value '/usr/lib/DataFlow*.o'
```

**Comment by oliverchang** [comment]
> Hi @inferno-chromium, do you have any ideas on the bazel rules_fuzzing integration failure in CI?
> 
> ```
> 	File "/root/.cache/bazel/_bazel_root/1f2d02b1c30820fa03d68ff064b1166c/external/rules_fuzzing/fuzzing/private/oss_fuzz/repository.bzl", line 41, column 17, in _extract_build_params
> 		fail("Unsupported $LIB_FUZZING_ENGINE value '%s'" % fuzzing_engine_library)
> Error in fail: Unsupported $LIB_FUZZING_ENGINE value '/usr/lib/DataFlow*.o'
> ```

@stefanbucur is this something that needs to be fixed in the bazel rules side?

That said, dataflow isn't well supported, so would be a good idea to remove this for now.

**Comment by stefanbucur** [comment]
> > Hi @inferno-chromium, do you have any ideas on the bazel rules_fuzzing integration failure in CI?
> > ```
> > 	File "/root/.cache/bazel/_bazel_root/1f2d02b1c30820fa03d68ff064b1166c/external/rules_fuzzing/fuzzing/private/oss_fuzz/repository.bzl", line 41, column 17, in _extract_build_params
> > 		fail("Unsupported $LIB_FUZZING_ENGINE value '%s'" % fuzzing_engine_library)
> > Error in fail: Unsupported $LIB_FUZZING_ENGINE value '/usr/lib/DataFlow*.o'
> > ```
> 
> @stefanbucur is this something that needs to be fixed in the bazel rules side?

We could adjust the code there, but I wanted to clarify first the syntax of `$LIB_FUZZING_ENGINE`. I was originally assuming it points to a static library to link in the fuzzing executable, so this is how the Bazel rules interpret that, too.

This one seems to be instead `/usr/lib/DataFlow*.o` - is that an unexpanded globbing pattern? How should users deal with `$LIB_FUZZING_ENGINE` in general?

**Comment by inferno-chromium** [comment]
yes just remove dataflow sanitizer please for now, unsupported completely.

**Comment by nareddyt** [comment]
> yes just remove dataflow sanitizer please for now, unsupported completely.

Done, thanks

**Comment by nareddyt** [comment]
cc @qiwzhang @taoxuy @justin-mp @shuoyang2016: As maintainers/contributors for the library, you will receive access to fuzz test crashes detected by OSS Fuzz. Feel free to filter out the emails if they are too noisy. I will monitor them.

**Comment by nareddyt** [comment]
Hi @inferno-chromium , can you merge this? Thanks

**Comment by nareddyt** [comment]
ping @inferno-chromium

**Comment by inferno-chromium** [comment]
I would let @asraa review and merge this. Also, can simplify this with https://github.com/google/oss-fuzz/tree/master/projects/bazel-rules-fuzzing-test helper, see new doc changes in https://google.github.io/oss-fuzz/getting-started/new-project-guide/bazel/#buildsh

File: projects/grpc-httpjson-transcoding/build.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# This project uses bazel rules_fuzzing. Script copied from OSS fuzz's bazel-rules-fuzzing-test.
```
Comment (asraa) [inline projects/grpc-httpjson-transcoding/build.sh:19]: Like @inferno-chromium said you could replace this with 
https://github.com/google/oss-fuzz/blob/008c7ec1f548755f2e97d5c0ed7e163a42ad32dd/infra/base-images/base-builder/bazel_build_fuzz_tests now!

**Review by asraa — COMMENTED** [review]
Other than the switch to use the macro, lgtm

File: projects/grpc-httpjson-transcoding/build.sh (hunk 2 above)
Comment (nareddyt) [reply projects/grpc-httpjson-transcoding/build.sh]: Done, thanks!

**Comment by nareddyt** [comment]
@asraa the build failure seems unrelated (?)

**Comment by asraa** [comment]
Ah, I think I merged the fix earlier, re-running won't help, can you try a main merge?

**Comment by nareddyt** [comment]
> Ah, I think I merged the fix earlier, re-running won't help, can you try a main merge?

Dpme

**Review by asraa — APPROVED** [review]

**Pull request merged**